### PR TITLE
Albert/login logout

### PR DESCRIPTION
**Pull request by 370417**
* User class now is better documented and has full functionality
* When you click the Sign In button in Login Activity, an event handler gets the username and password and does nothing with them yet

### CODE REVIEW

File: app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java [hunk 1]
```diff
@@ -2,12 +2,29 @@
...
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
 
 public class LoginActivity extends AppCompatActivity {
+    @BindView(R.id.username) EditText username;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java]: you should use access modifiers here. Also, please remember to use our conventional variable naming. eg. This should read: 
`@BindView(R.id.username) private EditText mUsernameEditText`; or `@BindView(R.id.username) private EditText mUsername`;

In the end, we want anyone who sees this variable called anywhere to know that it is a member variable (hence the "m" prefix) and should you choose to call it `mUsernameEditText` it also makes us know what it really is (an EditText which is a member variable)

File: app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java [hunk 2]
```diff
@@ -2,12 +2,29 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
+        ButterKnife.bind(this);
+    }
+
+    @OnClick(R.id.signin)
+    public void signin(View view) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java]: Will this method be called anywhere outside this activity? I doubt that. If it isn't, this should be marked `private`. Also use camel case here as such: `public void signIn(View view)`

File: app/src/main/java/edu/grinnell/appdev/grinnelldirectory/User.java [hunk 3]
```diff
@@ -1,37 +1,66 @@
...
     }
 
-    /* Sets new username and password if they are valid */
+    /**
+     * Save the username and password in shared preferences
+     * @param context context of the activity that calls this method
+     * @param username the user's new username
+     * @param password the user's new password
+     */
     public static void saveCredentials(Context context, String username, String password) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/appdev/grinnelldirectory/User.java:64]: Just to be on the safe side here, I'd just check to make sure `username` and `password` are not null or empty. Exit the function immediately if either or both are.

File: app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java [hunk 4]
```diff
@@ -2,12 +2,29 @@
 
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/appdev/grinnelldirectory/LoginActivity.java]: You've done some very good documentation in your methods and it'll be nice to extend that here too! Let's document this class :)

File: app/src/main/java/edu/grinnell/appdev/grinnelldirectory/User.java [hunk 5]
```diff
@@ -1,37 +1,66 @@
...
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.Editor;
 
 import android.content.Context;
 
+import com.google.gson.annotations.SerializedName;
+
 /**
- * Created by alsrfo on 2/10/17.
+ * The User class represents the user of the app.
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/appdev/grinnelldirectory/User.java:11]: I really like this documentation! Good job here, @as-f !

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Albert, this looks *very* good. A few changes and we'll be there! Please see my comments. I really like the work being done here and I hope others can look at this as well!

**Comment by 370417** [comment]
I made the changes @LarryAsante

**Review by LarryAsante — APPROVED** [review]

**Pull request merged**